### ISSUE TITLE
active directory authentication and temp table error

### ISSUE THREAD

**Issue by ginberg** [OHDSI/DatabaseConnector#278]
For Darwin we have a data partner that is using active directory authentication to setup their connection to sql server and they are getting an error when creating a temp table. In the error their UID is mentioned as schema name..
Is this connection setup supported by DatabaseConnector?

```
library(DatabaseConnector)
connectionDetails <- DatabaseConnector::createDbiConnectionDetails(dbms = "sql server",
                                                                   drv = odbc::odbc(),
                                                                   Driver = "ODBC Driver 18 for SQL Server",
                                                                   Server = Sys.getenv("SQL_SERVER"),
                                                                   Database = Sys.getenv("SQL_DB"),
                                                                   Authentication = "ActiveDirectoryPassword",
                                                                   UID = "EXT19616221@hustietoallas.fi",
                                                                   PWD = rstudioapi::askForPassword("OMOP Password"),
                                                                   pathToDriver = Sys.getenv("DRIVER_PATH"))
con <- connect(connectionDetails)
DatabaseConnector::insertTable(
  connection = con,
  tableName = "#mtcars",
  data = mtcars,
  dropTableIfExists = TRUE,
  createTable = TRUE,
  progressBar = FALSE,
  tempTable = TRUE,
  tempEmulationSchema = NULL,
  camelCaseToSnakeCase = TRUE
)
```
<img width="520" alt="error" src="https://github.com/user-attachments/assets/e3f2effd-49f7-4d0d-90ba-21c74f9e3966">

**Comment by salmarachidi** [comment]
<img width="557" alt="image" src="https://github.com/user-attachments/assets/14b3d12c-5816-45fd-8789-1923210826a0">
When I write down my password I get this error.

**Comment by schuemie** [comment]
@salmarachidi : this error message seems unrelated to how you specify the password. Perhaps your IT department can help you figure out the correct connection details when connecting using JDBC? (Such as the correct port)

**Comment by eric-fey-hus** [comment]
@schuemie We fixed the connection details, and it is showing "Connecting using SQL Server driver". But then error: "Failed to load MSAL4J Java library". When I try to fix this with jar files and jaddClassPath, I get other missing java libraries. Is there not a jar file that contains all the dependencies we need?

**Comment by ginberg** [comment]
@eric-fey-hus did you download the driver using the downloadJdbcDrivers method? See [this article](https://ohdsi.github.io/DatabaseConnector/articles/Connecting.html)

**Comment by eric-fey-hus** [comment]
Yes. I tried that earlier without success. Can try again tomorrow. But I would need a java 11 version. The download is java 8, at least the one I get.

**Comment by schuemie** [comment]
To my knowledge all JAR files should work with Java 8, but I certainly could have missed something.

I suspect MSAL4J is a library needed especially to enable the Active Directory authentication (See https://learn.microsoft.com/en-us/entra/msal/java/) . Adding this jar to the same place where the driver JAR files are should solve this problem. For the JDBC driver version that is installed by `downloadJdbcDrivers()` (version 9.2) it seems that should be enough: https://learn.microsoft.com/en-us/sql/connect/jdbc/connecting-using-azure-active-directory-authentication?view=sql-server-ver16#client-setup-requirements

**Comment by eric-fey-hus** [comment]
@schuemie @ginberg Trying to get the java dependencies to work became to complicated. Instead, I fixed the source code of DatabaseConnector. In function `insertTable.DatabaseConnectorDbiConnection` there was a line that removed the "#" from the the. This is what caused the issue, so I commented it out. Another fix, dbms(connection) should be `dbms(connection@dbiConnection)`, and it all worked out.  

I could create a branch on the DatabaseConnector repo to show you exactly what I did?

**Comment by schuemie** [comment]
Hi @eric-fey-hus. Glad you got it working! I would welcome a pull request. Would you mind forking the repo and create the PR from there?
